nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/AbstractTestCommentNotification.java

```diff
@@ -0,0 +1,172 @@
+
+        }, COMMENT_ADDED, DOCUMENT_CREATED);
+    }
+
+    @Test
+    public void shouldNotifyEventWhenUpdateComment() {
+        // We subscribe to the update document to check that we will not be notified about the comment updated as
+        // document (see CommentModificationVeto), only the comment updated.
+        Comment createdComment = createCommentAndAddSubscription("CommentUpdated", "Modification");
+
+        captureAndVerifyCommentEventNotification(() -> {
+            createdComment.setText("I update the message");
+            commentManager.updateComment(session, createdComment.getId(), createdComment);
+            return session.getDocument(new IdRef(createdComment.getId()));
+
+        }, COMMENT_UPDATED, DOCUMENT_UPDATED);
+    }
+
+    @Test
+    public void shouldNotifyEventWhenRemoveComment() {
+        Comment createdComment = createCommentAndAddSubscription("CommentRemoved");
+        DocumentModel commentDocModel = session.getDocument(new IdRef(createdComment.getId()));
+        commentDocModel.detach(true);
+
+        captureAndVerifyCommentEventNotification(() -> {
+            commentManager.deleteComment(session, createdComment.getId());
+            return commentDocModel;
+
+        }, COMMENT_REMOVED, DOCUMENT_REMOVED);
+    }
+
+    protected void captureAndVerifyCommentEventNotification(Supplier<DocumentModel> supplier, String commentEventType,
+            String documentEventType) {
+        try (CapturingEventListener listener = new CapturingEventListener(commentEventType, documentEventType)) {
+            DocumentModel commentDocumentModel = supplier.get();
+            transactionalFeature.nextTransaction();
+
+            assertTrue(listener.hasBeenFired(commentEventType));
+            assertTrue(listener.hasBeenFired(documentEventType));
+
+            List<Event> handledEvents = listener.getCapturedEvents()
+                                                .stream()
+                                                .filter(e -> commentEventType.equals(e.getName()))
+                                                .collect(Collectors.toList());
+
+            assertEquals(1, handledEvents.size());
+            Event expectedEvent = handledEvents.get(0);
+            assertEquals(commentEventType, expectedEvent.getName());
+
+            checkDocumentEventContext(expectedEvent, commentDocumentModel, commentedDocumentModel);
+        }
+    }
+
+    @Test
+    public void testCommentManagerType() {
+        assertEquals(getType(), commentManager.getClass());
+    }
+
+    protected Comment createCommentAndAddSubscription(String... notifications) {
+        NuxeoPrincipal principal = session.getPrincipal();
+        String subscriber = NotificationConstants.USER_PREFIX + principal.getName();
+        Arrays.stream(notifications)
+              .forEach(n -> notificationService.addSubscription(subscriber, n, commentedDocumentModel, false, principal,
+                      n));
```

Comment:
What about a simple loop here? same number of line and addSubscription call is on one line:
```
for (String notification : notifications) {
    notificationService.addSubscription(subscriber, n, commentedDocumentModel, false, principal, notification);
}
```